packages/flutter/test/cupertino/checkbox_test.dart

```diff
@@ -410,4 +461,305 @@ void main() {
+
+    const Color hoverColor = Color(0xFF000005);
+    const Color focusColor = Color(0xFF000006);
+    const Color checkColor = Color(0xffffffff);
+    const Color defaultActiveFillColor = Color(0xff007aff);
+    const Color defaultInactiveFillColor = Color(0xffffffff);
+    const Color pressedDarkShadow = Color(0x0d000000);
+
+    Color getFillColor(Set<WidgetState> states) {
+      if (states.contains(WidgetState.hovered)) {
+        return hoverColor;
+      }
+      if (states.contains(WidgetState.focused)) {
+        return focusColor;
+      }
+      if (states.contains(WidgetState.selected)){
+        return defaultActiveFillColor;
+      }
+      return defaultInactiveFillColor;
+    }
+
+    await tester.pumpWidget(buildApp(
+      fillColor: WidgetStateProperty.resolveWith(getFillColor),
+      value: false
+    ));
+    final TestGesture gesture1 =
+      await tester.startGesture(tester.getCenter(find.byType(CupertinoCheckbox)));
+    await tester.pumpAndSettle();
+
+    expect(
+      find.byType(CupertinoCheckbox),
+      paints
+        ..path(color: defaultInactiveFillColor)
+        ..drrect()
+        ..path(color: pressedDarkShadow),
+      reason: 'Inactive pressed Checkbox is slightly darkened',
+    );
+
+    await tester.pumpWidget(buildApp(
+        fillColor: WidgetStateProperty.resolveWith(getFillColor),
+      ));
+    final TestGesture gesture2 =
+      await tester.startGesture(tester.getCenter(find.byType(CupertinoCheckbox)));
+    await tester.pumpAndSettle();
+
+    expect(
+      find.byType(CupertinoCheckbox),
+      paints
+        ..path(color: defaultActiveFillColor)
+        ..rrect()
+        ..path(color: checkColor)
+        ..path(color: checkColor)
+        ..path(color: pressedDarkShadow),
+      reason: 'Active pressed Checkbox is slightly darkened',
+    );
+
+    await tester.pumpWidget(Container()); // reset test
+    await tester.pumpWidget(buildApp(
+      autofocus: true,
+      focusNode: focusNode,
+      fillColor: WidgetStateProperty.resolveWith(getFillColor))
+    );
+    await tester.pumpAndSettle();
+
+    expect(focusNode.hasPrimaryFocus, isTrue);
+    expect(
+      find.byType(CupertinoCheckbox),
+      paints
+        ..path(color: focusColor),
+      reason: 'Focused Checkbox should use $focusColor',
+    );
+
+    // Start hovering
+    final TestGesture gesture5 =
+      await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await gesture5.addPointer();
+    addTearDown(gesture5.removePointer);
+    await gesture5.moveTo(tester.getCenter(find.byType(CupertinoCheckbox)));
+    await tester.pumpAndSettle();
+
+    expect(
+      find.byType(CupertinoCheckbox),
+      paints
+        ..path(color: hoverColor),
+      reason: 'Hovered Checkbox should use $hoverColor',
+    );
+
+    // Finish gestures to release resources.
+    await gesture1.up();
+    await gesture2.up();
+    await tester.pumpAndSettle();
+  });
```

Comment:
Are most of these tests ported from Material's Checkbox tests?

Reply:
Yes. They are basically the same new tests added in the previous adaptive checkbox PR.